File: BE/src/main/java/team29/hoorry/issuetracker/core/issue/dto/IssueFilter.java
package team29.hoorry.issuetracker.core.issue.dto;

import java.util.List;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.util.MultiValueMap;
import team29.hoorry.issuetracker.core.common.search.SearchFilter;
import team29.hoorry.issuetracker.core.issue.domain.Status;

@Getter
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@ToString
public class IssueFilter {

	private List<String> labelTitles;
	private String milestoneTitle;
	private List<String> assigneeNames;
	private String writerName;
	private String searchParam;
	private Status status;

	public static IssueFilter from(MultiValueMap<SearchFilter, String> filters) {
		IssueFilter issueFilter = new IssueFilter();

		issueFilter.labelTitles = filters.get(SearchFilter.LABEL_TITLE);
		issueFilter.milestoneTitle = filters.getFirst(SearchFilter.MILESTONE_TITLE);
		issueFilter.assigneeNames = filters.get(SearchFilter.ASSIGNEE_NAME);
		issueFilter.writerName = filters.getFirst(SearchFilter.WRITER_NAME);
		issueFilter.searchParam = filters.getFirst(SearchFilter.SEARCH_PARAM);
		String status = filters.getFirst(SearchFilter.STATUS);
		if ("open".equals(status)) {
			issueFilter.status= Status.OPEN;
		}
		if ("closed".equals(status)) {
			issueFilter.status = Status.CLOSED;
		}

		return issueFilter;
	}
}

File: fe/src/helpers/constants/IssueFilter.ts
const ISSUE_FILTER = [
  {
    id: 'open',
    title: '열린 이슈',
    type: 'status',
  },
  {
    id: '@me',
    title: '내가 작성한 이슈',
    type: 'author',
  },
  {
    id: '@me',
    title: '나에게 할당된 이슈',
    type: 'assignee',
  },
  {
    id: '@me',
    title: '내가 댓글을 남긴 이슈',
    type: 'mentions',
  },
  {
    id: 'closed',
    title: '닫힌 이슈',
    type: 'status',
  },
];

export default ISSUE_FILTER;
